# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1362139532 18000
#      Fri Mar 01 07:05:32 2013 -0500
# Node ID fbc8f7afb1931fd92d903ea6af8c99a3d8d232cc
# Parent  c5bfdc4c09632e1a392a49e2d357040ce911e774
* input.cc (get_user_input): Only return eof if retval is empty.

diff --git a/libinterp/interpfcn/input.cc b/libinterp/interpfcn/input.cc
--- a/libinterp/interpfcn/input.cc
+++ b/libinterp/interpfcn/input.cc
@@ -355,17 +355,18 @@ get_user_input (bool& eof)
   if (get_input_from_eval_string)
     {
       retval = current_eval_string;
 
       // Clear the global eval string so that the next call will return
       // an empty character string with EOF = true.
       current_eval_string = "";
 
-      eof = true;
+      if (retval.empty ())
+        eof = true;
     }
   else
     retval = octave_gets (eof);
 
   current_input_line = retval;
 
   return retval;
 }
